src/core/ext/filters/fault_injection/fault_injection_filter.cc

```diff
@@ -428,6 +428,7 @@ void CallData::DelayBatch(grpc_call_element* elem,
   MutexLock lock(&delay_mu_);
   delayed_batch_ = batch;
   resume_batch_canceller_ = new ResumeBatchCanceller(elem);
+  ExecCtx::Get()->InvalidateNow();
```

Comment:
Please add a comment here explaining why this is needed.

Reply:
Comments added:

```
  // By default, ExecCtx::Get()->Now() returns a cached timestamp. If there are
  // thousands of RPCs happen on one thread, we might observe ms-level error in
  // Now(). This could mean the construction of RPC object is microseconds
  // earlier than the filter execution. But we still haven't found the root
  // cause. Read more: https://github.com/grpc/grpc/pull/25738.
```